### ISSUE TITLE
autocomplete ajax source

### ISSUE THREAD

**Issue by lyrad** [vedmack/yadcf#250]
Hi Daniel,
I'm tying to fill an yadcf jqueryui autocomplete with ajax data :
```javascript
var table_details_yadcf_init = [ {
    	    column_number: 1,
    	    filter_default_label: "Saisissez...",
    	    filter_container_id: "table_details_filters_1",
    	    filter_type: "auto_complete",
    	    filter_plugin_options: {
    	    	source: function( request, response ) {
    	            $.ajax( {
    	                url: "/referentiel/searchequipementlieulibelle",
    	                dataType: "json",
    	                data: { q: request.term },
    	                success: function( data ) { response( data.a ); }
    	            } );
    	        },
    	        minLength: 4
    	    }
    	} ];
```

But look like YADCF doesnt want let me use a function as "source attribute jqueryui autocomplete" (line 2615 0.9.22 beta) : 
```javascript
if (columnObj.filter_plugin_options !== undefined) {
    columnObj.filter_plugin_options.source = $(document).data("yadcf-filter-" + table_selector_jq_friendly + "-" + column_number);
    columnObj.filter_plugin_options.select = autocompleteSelect;
} else {
    columnObj.filter_plugin_options = {
        source: $(document).data("yadcf-filter-" + table_selector_jq_friendly + "-" + column_number),
	select: autocompleteSelect
	};
}
```
I guess .data() get the "yadcf_data_x" array back from the server or defined as html_data_selector / data.....

So I added an if, look like it works... 
```javascript
if(columnObj.filter_plugin_options.source === undefined) {
    columnObj.filter_plugin_options.source = $(document).data("yadcf-filter-" + table_selector_jq_friendly + "-" + column_number);
}
```

If that doesnt make any trouble with others functionnality, could you add something like that in 0.9? 
Thanks!

**Comment by vedmack** [comment]
Hi,

I commited a quick fix, grab [0.8.9.beta.23](https://github.com/vedmack/yadcf/blob/09d90ef8629590b8403bcad682e71bdd9999cd8f/jquery.dataTables.yadcf.js) and look how it works for you


Updated code looks like this:

```
if (columnObj.filter_plugin_options !== undefined) {
	if (columnObj.filter_plugin_options.source === undefined) {
		columnObj.filter_plugin_options.source = $(document).data("yadcf-filter-" + table_selector_jq_friendly + "-" + column_number);
	}
	columnObj.filter_plugin_options.select = autocompleteSelect;
}
```

**Comment by lyrad** [comment]
Thank you!
